BCM2710_DT: fix gpio expander bindings

vianpl opened the pull request:
The upstreamed driver for the GPIO expander expects to be a children of
the "firmware" node. It won't probe otherwise.

The patch also removes the "firmware" phandle as it's useless.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>

Commit:
BCM2710_DT: fix gpio expander bindings

The upstreamed driver for the GPIO expander expects to be a children of
the "firmware" node.

The patch also removes the "firmware" phandle as it's useless.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>

vianpl commented:
If it's fine by you, it would also be nice to have in rpi-4.19.y.

pelwell commented:
Have you tried cherry picking to 4.19? In a brief end-of-day test it seemed to break the expander, but I didn't have time to investigate.

vianpl commented:
Sorry my bad, I missed a commit, I'll update it and test it tomorrow

Commit:
BCM2708_DT: update firmware node binding

The upstreamed version of the firmware node has been updated to present
it as a "simple-bus". We need to get this in order to accomodate other
device bindings, namely RPi's firmware based gpio expander.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>

vianpl commented:
Updated the pull request. Tested with raspbian + 4.19.4-v7, exp-gpio comes up properly.

pelwell commented:
Thanks. The updated PR has been merged to rpi-4.19.y as well.

vianpl commented:
Thanks! :+1:
